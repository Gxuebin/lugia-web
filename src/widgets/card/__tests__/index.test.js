/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';

import 'jest-styled-components';
import Wrapper from '../demo';
import renderer from 'react-test-renderer';

describe('Card', () => {
  it('css', () => {
    const Target = <Wrapper />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
});
