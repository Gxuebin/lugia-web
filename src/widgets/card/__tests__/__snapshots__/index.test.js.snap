// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card css 1`] = `
.c30 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  font-size: 1.2rem;
}

.c13 {
  font-variant: tabular-nums;
  color: rgba(0,0,0,0.65);
  box-sizing: border-box;
  display: inline-block;
  text-align: center;
  background: #cccccc;
  white-space: nowrap;
  position: relative;
  width: 6.666666666666667em;
  height: 6.666666666666667em;
  line-height: 2.666666666666667em;
  border-radius: 50%;
}

.c14 {
  border-radius: 50%;
  width: 6.666666666666667em;
  height: 6.666666666666667em;
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 30em;
  height: 14.000000000000002em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 25em;
  height: 16.666666666666668em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 12.5em;
  height: 19em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0.4166666666666667em;
}

.c16 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 27.5em;
  height: 12.833333333333334em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c19 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 17.5em;
  height: 22.333333333333336em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0.4166666666666667em;
}

.c22 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 27.5em;
  height: 12.5em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c23 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 30em;
  height: 20em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c40 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 16.666666666666668em;
  height: 18.333333333333336em;
  box-shadow: 0 0 0.5em rgba(0,0,50,0.1);
  background: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 0.4166666666666667em;
}

.c3 {
  margin-left: 0.8333333333333334em;
  font-size: 1.2rem;
  margin-bottom: 1em;
  margin-top: 2.166666666666667em;
  display: inline-block;
  width: inherit;
  height: inherit;
}

.c7 {
  margin-left: 2.5em;
  font-size: 1.2rem;
  margin-bottom: 1em;
  margin-top: 2.166666666666667em;
  display: inline-block;
  width: inherit;
  height: inherit;
}

.c15 {
  margin-left: 0em;
  font-size: 1.2rem;
  text-align: center;
  display: inline-block;
  width: inherit;
  height: inherit;
}

.c24 {
  margin-left: 0em;
  font-size: 1.2rem;
  margin-bottom: 1em;
  margin-top: 2.166666666666667em;
  display: inline-block;
  width: inherit;
  height: inherit;
}

.c21 {
  width: inherit;
  height: inherit;
}

.c11 {
  position: relative;
  width: inherit;
  height: 9.333333333333334em;
  display: block;
  text-align: center;
}

.c17 {
  position: relative;
  width: 10em;
  height: inherit;
  display: inline-block;
  text-align: center;
}

.c20 {
  position: relative;
  width: 16.666666666666668em;
  height: 10.833333333333334em;
  display: block;
  text-align: center;
}

.c12 {
  margin: 1.6666666666666667em 0;
  border-radius: 50%;
}

.c18 {
  margin: 1.6666666666666667em 1.6666666666666667em;
  border-radius: 50%;
}

.c4 {
  text-align: inherit;
  display: inline-block;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.6rem;
  color: #333333;
  font-weight: 500;
  overflow: hidden;
  white-space: nowrap;
}

.c9 {
  text-align: inherit;
  display: inline-block;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.6rem;
  color: #333333;
  font-weight: 700;
  overflow: hidden;
  white-space: nowrap;
}

.c8 {
  left: 1.1666666666666667em;
  position: absolute;
  display: inline-block;
  height: 1.6666666666666667em;
  width: 0.4166666666666667em;
  background: #4d63ff;
  border-radius: 0.4166666666666667em;
}

.c5 {
  text-align: inherit;
  font-size: 1.4rem;
  margin-top: 0.33333333333333337em;
  color: #666666;
}

.c2 {
  position: absolute;
  font-size: 1.4rem;
  font-weight: 400;
  right: 1.6666666666666667em;
  top: 1.3333333333333335em;
}

.c33 {
  position: relative;
  cursor: pointer;
  white-space: nowrap;
  cursor: pointer;
  display: inline-block;
  line-height: 2.8333333333333335em;
  height: 2.8333333333333335em;
  padding: 0 1.6666666666666667em 0 1.6666666666666667em;
  box-shadow: 0 0 0.5em none;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  background: none;
  z-index: 5;
  box-sizing: border-box;
}

.c33:hover > div {
  color: #4d63ff;
}

.c33:hover > span > i:first-child {
  color: #4d63ff;
}

.c34 {
  color: #4d63ff;
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
  padding: 0 0em 0 0em;
  height: 2.8333333333333335em;
  line-height: 2.8333333333333335em;
}

.c34:focus {
  color: #4d63ff;
}

.c35 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
  padding: 0 0em 0 0em;
  height: 2.8333333333333335em;
  line-height: 2.8333333333333335em;
}

.c35:focus {
  color: #4d63ff;
}

.c39 {
  display: inline-block;
  overflow: hidden;
  width: 100%;
}

.c36 {
  background-color: blue;
  position: absolute;
  box-sizing: border-box;
  z-index: 3;
  border-radius: 0.16666666666666669em;
  height: 0.16666666666666669em;
  width: 8.333333333333334em;
  bottom: 0;
  -webkit-transform: translateX(1.6666666666666667em);
  -ms-transform: translateX(1.6666666666666667em);
  transform: translateX(1.6666666666666667em);
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c28 {
  position: absolute;
  font-size: 1.2rem;
  display: none;
  z-index: 5;
  cursor: not-allowed;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  line-height: 100%;
  top: 50%;
  left: 0.8333333333333334em;
}

.c37 {
  position: absolute;
  font-size: 1.2rem;
  display: none;
  z-index: 5;
  cursor: pointer;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  line-height: 100%;
  top: 50%;
  right: 0.8333333333333334em;
}

.c26 {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  position: relative;
  width: NaNem;
  height: NaNem;
}

.c31 {
  width: 100%;
  display: inline-block;
  white-space: nowrap;
  overflow: hidden;
}

.c32 {
  height: 2.8333333333333335em;
  display: inline-block;
  box-sizing: border-box;
  white-space: nowrap;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  -webkit-transform: translateX(0em);
  -ms-transform: translateX(0em);
  transform: translateX(0em);
}

.c27 {
  height: 2.8333333333333335em;
  line-height: 2.8333333333333335em;
  position: relative;
  border-bottom: 0.08333333333333334em solid #e8e8e8;
  background: none;
  margin: 1.6666666666666667em 0;
  z-index: 99;
}

.c29 {
  display: inline-block;
  font-style: normal;
  text-align: center;
  font-size: 1rem;
  cursor: not-allowed;
}

.c38 {
  display: inline-block;
  font-style: normal;
  text-align: center;
  font-size: 1rem;
  cursor: pointer;
}

.c0 {
  margin-left: 50px;
  margin-top: 50px;
  display: inline-block;
}

.c25 {
  display: inline-block;
  margin-left: 20px;
}

.c41 {
  text-align: center;
  width: 100%;
  height: 100%;
}

.c42 {
  text-align: center;
  font-size: 16px;
  color: #666;
  margin-bottom: 26px;
  width: 100%;
}

.c43 {
  text-align: center;
  color: #4d63ff;
  display: inline-block;
}

.c44 {
  position: absolute;
  text-align: center;
  font-size: 14px;
  border-radius: 20px;
  margin-top: 32px;
  color: #fff;
  bottom: 10%;
  width: 80%;
  background-color: #4d63ff;
  height: 32px;
  border: 2px solid #4d63ff;
  left: 10%;
}

<div
  className="c0"
>
  <p>
    基本样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c1"
      type="simple"
    >
      <div
        className="c2"
      >
        操作
      </div>
      <div
        className="c3"
        type="simple"
      >
        <div
          className="c4"
          type="simple"
        >
          this is title
           
        </div>
        <div
          className="c5"
        >
          this is description
           
        </div>
      </div>
    </div>
  </div>
  <p>
    基本样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c6"
      type="simple"
    >
      <div
        className="c3"
        type="simple"
      >
        <div
          className="c4"
          type="simple"
        >
          <div>
            this is title
          </div>
           
        </div>
        <div
          className="c5"
        >
          <div>
            this is description
          </div>
          <div>
            this is description
          </div>
          <div>
            this is description
          </div>
          <div>
            this is description
          </div>
           
        </div>
      </div>
    </div>
  </div>
  <p>
    tip样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c6"
      type="tip"
    >
      <div
        className="c7"
        content={
          Array [
            <div>
              this is description
            </div>,
            <div>
              this is description
            </div>,
          ]
        }
        type="tip"
      >
        <div
          className="c8"
        />
        <div
          className="c9"
          type="tip"
        >
          <div>
            this is title
          </div>
           
        </div>
        <div>
          this is description
        </div>
        <div>
          this is description
        </div>
      </div>
    </div>
  </div>
  <p>
    children样式 
  </p>
  <div
    className="c0"
  >
    <div
      className="c1"
      type="simple"
    >
      <div
        className="c3"
        type="simple"
      >
        <div>
          children 样式的卡片
        </div>
      </div>
    </div>
  </div>
  <p>
    头像样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c10"
      type="avatar"
    >
      <div
        className="c11"
        size={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
        type="avatar"
      >
        <span
          className="c12 c13"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          shape="circle"
          size="default"
          src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/4d4bd6db04b1c6015acf4c933607956a9f2d62a1/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/Bitmap2.png"
        >
          <img
            className="c14"
            shape="circle"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/4d4bd6db04b1c6015acf4c933607956a9f2d62a1/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/Bitmap2.png"
          />
        </span>
      </div>
      <div
        className="c15"
        type="avatar"
      >
        <div
          className="c4"
          type="avatar"
        >
          this is title
           
        </div>
        <div
          className="c5"
        >
          this is description
           
        </div>
      </div>
    </div>
  </div>
  <p>
    头像样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c16"
      type="avatar"
    >
      <div
        className="c17"
        size={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
        type="avatar"
      >
        <span
          className="c18 c13"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          shape="circle"
          size="default"
          src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/4d4bd6db04b1c6015acf4c933607956a9f2d62a1/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/Bitmap2.png"
        >
          <img
            className="c14"
            shape="circle"
            src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/4d4bd6db04b1c6015acf4c933607956a9f2d62a1/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/Bitmap2.png"
          />
        </span>
      </div>
      <div
        className="c3"
        type="avatar"
      >
        <div
          className="c4"
          type="avatar"
        >
          this is title
           
        </div>
        <div
          className="c5"
        >
          <div>
            <div>
              this is description
            </div>
            <div>
              this is description
            </div>
          </div>
           
        </div>
      </div>
    </div>
  </div>
  <p>
    图片样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c19"
      type="image"
    >
      <div
        className="c20"
        size={
          Object {
            "height": 130,
            "width": 200,
          }
        }
        type="image"
      >
        <img
          className="c21"
          src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/2eac1a340185301d24d6fac426aebd9abe6dea0e/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/18081548404150_.pic_hd.jpg"
        />
      </div>
      <div
        className="c3"
        type="image"
      >
        <div
          className="c4"
          type="image"
        >
          this is title
           
        </div>
        <div
          className="c5"
        >
          this is description
           
        </div>
      </div>
    </div>
  </div>
  <p>
    图片样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c22"
      type="image"
    >
      <div
        className="c17"
        size={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
        type="image"
      >
        <img
          className="c21"
          src="http://192.168.102.73:8081/BigFrontend/Work/ued/lugia/raw/2eac1a340185301d24d6fac426aebd9abe6dea0e/lugiaweb%E7%BB%84%E4%BB%B6/%E5%8D%A1%E7%89%87/18081548404150_.pic_hd.jpg"
        />
      </div>
      <div
        className="c3"
        type="image"
      >
        <div
          className="c4"
          type="image"
        >
          this is title
           
        </div>
        <div
          className="c5"
        >
          this is description
           
        </div>
      </div>
    </div>
  </div>
  <p>
    组合样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c23"
      type="combo"
    >
      <div
        className="c2"
      >
        操作
      </div>
      <div
        className="c24"
        content={
          <styled.div>
             
            <lugia_widget_ThemeWrapWidgetlugia_widget_Tabs
              data={
                Array [
                  Object {
                    "content": "content of Tab1",
                    "title": "Tab1",
                  },
                  Object {
                    "content": "content of Tab2",
                    "title": "Tab2",
                  },
                  Object {
                    "content": "content of Tab3",
                    "title": "Tab3",
                  },
                ]
              }
            />
          </styled.div>
        }
        type="combo"
      >
        <div
          className="c25"
        >
           
          <div
            className="c26"
          >
            <div
              className="c27"
            >
              <span
                className="c28"
                onClick={[Function]}
                type="pre"
              >
                <i
                  className="lugia-icon-direction_Left c29 c30"
                />
              </span>
              <div
                className="c31"
              >
                <div
                  className="c32"
                  x={-0}
                >
                  <div
                    className="c33"
                    disabled={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <div
                      className="c34"
                      disabled={false}
                    >
                      Tab1
                    </div>
                  </div>
                  <div
                    className="c33"
                    disabled={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <div
                      className="c35"
                      disabled={false}
                    >
                      Tab2
                    </div>
                  </div>
                  <div
                    className="c33"
                    disabled={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <div
                      className="c35"
                      disabled={false}
                    >
                      Tab3
                    </div>
                  </div>
                  <div
                    className="c36"
                    x={20}
                  />
                </div>
              </div>
              <span
                className="c37"
                onClick={[Function]}
                type="next"
              >
                <i
                  className="lugia-icon-direction_right c38 c30"
                />
              </span>
            </div>
            <div
              className="c39"
            >
              content of Tab1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <p>
    组合样式
  </p>
  <div
    className="c0"
  >
    <div
      className="c40"
      type="combo"
    >
      <div
        className="c24"
        content={<AmountCard />}
        type="combo"
      >
        <div
          className="c41"
        >
          <div
            className="c42"
          >
            价格
          </div>
          <div
            className="c43"
          >
            <div
              className="c43"
              style={
                Object {
                  "fontSize": 32,
                }
              }
            >
              9
            </div>
            <div
              className="c43"
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              .88
            </div>
          </div>
          <button
            className="c44"
          >
            购买
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
